build-tools/scripts/runtime-helpers.mk

```diff
@@ -0,0 +1,91 @@
+#
+# Various helper/shortcut targets
+#
+MONO_RUNTIMES_DIR = build-tools/mono-runtimes
+MONO_RUNTIMES_PROJECT = $(MONO_RUNTIMES_DIR)/mono-runtimes.mdproj
+MONO_RUNTIMES_BUILD_DIR = $(MONO_RUNTIMES_DIR)/obj/$(CONFIGURATION)
+
+# $(1) - architecture name
+define CreateMonoRebuildRuleForArch
+rebuild-$(1)-mono: touch-mono-head
+	@if ! test -d $$(MONO_RUNTIMES_BUILD_DIR)/$(1); then \
+		echo $$(MONO_RUNTIMES_BUILD_DIR)/$(1) ; \
+		echo Runtime for architecture $(1) and configuration $$(CONFIGURATION) was not built yet or is not enabled in configuration or cached Mono runtime was installed ; \
+		echo In order to build Mono for $(1) please enable it in Configuration.Override.props file and ; \
+		echoinfo Run the following command: make rebuild-mono ; \
+		exit 1 ; \
+	fi
+	$$(MAKE) -C $$(MONO_RUNTIMES_BUILD_DIR)/$(1)
+	$$(MSBUILD) /t:_InstallRuntimes $$(MONO_RUNTIMES_PROJECT)
+
+.PHONY: rebuild-$(1)-mono
+endef
+
+#
+# This is less than ideal, but that's how things are right now.
+# Do NOT push the change to Mono!
+#
+touch-mono-head:
+	@(cd external/mono; git commit --amend -C HEAD --date="`date`")
+
+rebuild-mono: rebuild-all-mono
+rebuild-all-mono: touch-mono-head
+	$(MSBUILD) /t:ForceBuild $(MONO_RUNTIMES_PROJECT)
+
+$(foreach arch,$(ALL_JIT_ABIS),$(eval $(call CreateMonoRebuildRuleForArch,$(arch))))
+
+.PHONY: rebuild-mono rebuild-all-mono touch-mono-head
+
+rebuild-bcl-assembly:
+	@if [ -z "$(ASSEMBLY)" ]; then \
+		echo Assembly name required. Run make as follows: ; \
+		echo $(MAKE) ASSEMBLY=bcl_assembly_name ; \
+		exit 1 ; \
+	fi
+	$(MAKE) -C external/mono/mcs/class/$(ASSEMBLY)/ PROFILE=monodroid
+	$(MSBUILD) /t:_InstallBcl $(MONO_RUNTIMES_PROJECT)
+
+rebuild-all-bcl:
+	@if [ ! -d $(MONO_RUNTIMES_BUILD_DIR)/host-$(OS) ]; then \
+		echo Host Mono runtime for $(OS) has not been built ; \
+		echo or cached Mono runtime was installed ; \
+		echo In order to build host Mono for $(OS) please enable it in Configuration.Override.props file and ; \
+		echo Run the following command: make rebuild-mono ; \
+		echo Unable to rebuild all BCL assemblies ; \
+	fi
+	make -C $(MONO_RUNTIMES_BUILD_DIR)/host-$(OS)
+
+.PHONY: rebuild-bcl-assembly rebuild-all-bcl
+
+#
+# $(1) - arch name
+#
+# NOTE: the first empty line must stay since Make displays it verbatim...
+#
+define MonoArchRebuildHelp
+
+	@echo "  rebuild-$(1)-mono"
+	@echo "     Rebuild and install Mono runtime for the $(1) architecture only regardless"
+	@echo "     of whether a cached copy was used."
+	@echo
+endef
+
+rebuild-help:
+	@echo Helper targets to rebuild the Mono runtime and BCL assemblies
+	@echo
+	@echo "  rebuild-mono"
```

Comment:
I'd call this `rebuild-mono-runtime` to make it clear that it only builds the runtime.